test/inductor/test_torchinductor_dynamic_shapes.py

```diff
@@ -368,6 +369,45 @@ def f(x):
         arg = torch.tensor(5, device=device)
         self.assertEqual(f(arg), cf(arg))
 
+    @torch._dynamo.config.patch(capture_scalar_outputs=True, capture_dynamic_output_shape_ops=True)
+    @torch._inductor.config.patch(implicit_fallbacks=True)
+    def test_unbacked_save_for_backwards(self, device) -> None:
+        @torch.library.custom_op("_test::_cat", mutates_args=())
+        def _cat(t: torch.Tensor, ds: List[int]) -> torch.Tensor:
+            return t * t.new_ones([sum(ds)])
```

Comment:
We need a custom cat, because regular cat backwards forces u0/u1 to get saved for backwards so you don't trigger the bug.